src/libostree/ostree-repo.h

```diff
@@ -969,7 +997,9 @@ typedef struct {
   OstreeRepoDevInoCache *devino_to_csum_cache;
 
   int unused_ints[6];
-  gpointer unused_ptrs[5];
+  gpointer unused_ptrs[3];
+  OstreeRepoCheckoutFilter filter; /* Since: 2018.2 */
+  gpointer filter_user_data; /* Since: 2018.2 */
```

Comment:
I can transform those two into a single object, similar to `OstreeRepoCommitModifier`, if we want to conserve pointers.

Reply:
Eh, feels unlikely we keep adding a lot more functionality to checkout.  If we do, I think I'd vote we add new high level APIs.